### PR TITLE
zec: Satisfy WalletHistorian.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Copies a lot of code from #2748

### CODE REVIEW

**Comment by JoeGruffins** [comment]
It looks like the harness loadbot is failing with `2024-06-04 18:07:27.051 [ERR] LOADBOT: exec error (zec) "./connect-alpha 41193": exit status 1: Error reading configuration file: Missing zcash.conf` I don't think related to these changes.

File: client/asset/zec/zec.go [hunk 1]
```diff
@@ -2912,3 +3010,679 @@ func deserializeTx(b []byte) (*wire.MsgTx, error) {
...
+
+func (w *zecWallet) startTxHistoryDB(ctx context.Context) (*sync.WaitGroup, error) {
+	addressPath, err := w.findExistingAddressBasedTxHistoryDB()
+	if err != nil {
+		return nil, err
+	}
+	dbPath := addressPath
+
+	if dbPath == "" {
+		depositAddr, err := w.transparentAddress()
```
Comment (buck54321) [inline client/asset/zec/zec.go:3162]: Can we use the `mnemonic_seedfp` parameter of the[ `getwalletinfo` result](https://zcash.github.io/rpc/getwalletinfo.html).

**Review by buck54321 — CHANGES_REQUESTED** [review]

File: client/asset/zec/transparent_rpc.go [hunk 2]
```diff
@@ -38,12 +38,25 @@ func lockUnspent(c rpcCaller, unlock bool, ops []*btc.Output) error {
 	return err
 }
 
-func getTransaction(c rpcCaller, txHash *chainhash.Hash) (*dexzec.Tx, error) {
+type zTx struct {
+	*dexzec.Tx
+	blockHash string
+}
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/asset/zec/transparent_rpc.go:44]: Ok.

File: client/asset/zec/zec.go [hunk 3]
```diff
@@ -46,6 +49,11 @@ const (
 	minNetworkVersion   = 5040250 // v5.4.2
 	walletTypeRPC       = "zcashdRPC"
 
+	// defaultRedeemConfTarget is the default redeem transaction confirmation
+	// target in blocks used by estimatesmartfee to get the optimal fee for a
+	// redeem transaction.
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/asset/zec/zec.go:54]: Changed to `defaultConfTarget`

**Comment by JoeGruffins** [comment]
Using the encoded seed as the name and other things https://github.com/decred/dcrdex/compare/3c0fd17b16f9cc181b988ea018000502604bee05..352ba72b28a49c01524189d8ea68fa93bbe433af

loadbot is working for zec and zcl now, but zec fails rather quickly with:
```
2024-06-17 14:59:18.835 [DBG] CORE:STACKER:0[zec]: Attempting to fund Zcash order, maxFeeRate = 200, max swaps = 4
2024-06-17 14:59:18.842 [DBG] CORE:STACKER:0[zec]: Funding 0.04 BTC ZEC order with coins [50b3df8e186158f3377ec331b66fd4172ee504af6b9cd3bb54cecb66fef64f58:0] worth 1.2875 BTC
2024-06-17 14:59:18.847 [DBG] CORE:STACKER:0[zec]: Unlocked spent outpoint 50b3df8e186158f3377ec331b66fd4172ee504af6b9cd3bb54cecb66fef64f58:0
2024-06-17 14:59:18.850 [INF] LOADBOT[STACKER:0]: Balance: zec = 8739920000 available, 2703750000 locked
2024-06-17 14:59:18.850 [DBG] CORE:STACKER:0[zec]: returning coins [50b3df8e186158f3377ec331b66fd4172ee504af6b9cd3bb54cecb66fef64f58:0]
2024-06-17 14:59:18.851 [CRT] MANTLE:STACKER:0: Trade error (limit order, sell = true, qty = 4000000, rate = 83200000): new order request with DEX server 127.0.0.1:17273 market zec_btc failed: rpc error: error code 36: coin 50b3df8e186158f3377ec331b66fd4172ee504af6b9cd3bb54cecb66fef64f58:0 is locked
```

File: client/asset/zec/zec.go [hunk 4]
```diff
@@ -2912,3 +3010,679 @@ func deserializeTx(b []byte) (*wire.MsgTx, error) {
...
+	}
+
+	txs, err := w.TxHistory(1, &txID, false)
+	if err != nil {
+		return nil, err
+	}
+	if len(txs) == 0 {
+		return nil, asset.CoinNotFoundError
+	}
+
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/asset/zec/zec.go]: Doing that.

**Pull request merged**